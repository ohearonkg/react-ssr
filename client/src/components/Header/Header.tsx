import * as React from "react";

const wrapperStyles = {
  width: "100%",
  display: "flex",
  alignItems: "center"
};

const navItemsWrapperStyles = {
  flex: "1 1 auto",
  display: "flex",
  justifyContent: "flex-end",
  color: "black"
};

const navLinkStyles = {
  textDecoration: "none",
  padding: "5px"
};

export default () => (
  <div style={wrapperStyles}>
    <div>React SSR</div>
    <nav style={navItemsWrapperStyles}>
      <a style={navLinkStyles} href="#">
        Home
      </a>
      <a style={navLinkStyles} href="#">
        User List
      </a>
      <a style={navLinkStyles} href="#">
        Admin List
      </a>
    </nav>
  </div>
);
