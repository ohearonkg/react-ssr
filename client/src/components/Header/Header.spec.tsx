import { shallow, ShallowWrapper } from "enzyme";
import toJson from "enzyme-to-json";
import * as React from "react";
import Header from "./Header";

describe("Header Component", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<Header />);
  });

  /**
   * Snapshot
   */
  it("Should match its snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  /**
   * Render login button
   */
  it("should render a login button if its prop value of isAuthenticated is false", () => {
    wrapper.setProps({ isAuthenticated: false });
    expect(wrapper.html()).toMatch("Login");
  });
});
