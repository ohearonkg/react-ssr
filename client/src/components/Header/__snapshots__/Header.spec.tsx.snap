// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Should match its snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "width": "100%",
    }
  }
>
  <div>
    React SSR
  </div>
</div>
`;
